ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos/Common.hs

```diff
@@ -54,36 +55,178 @@ data PraosChainSelectView c = PraosChainSelectView
+--
+-- 1. Chain length, with longer chains always preferred.
+--
+-- 2. If the tip of each chain was issued by the same agent, then we prefer the
+--    candidate if it has a higher ocert issue number.
+--
+-- 3. Depending on the 'VRFTiebreakerFlavor':
+--
+--     * If 'UnrestrictedVRFTiebreaker': Compare via a VRF value from the chain
+--       tip, with lower values preferred. See @pTieBreakVRFValue@ for which one
+--       is used.
+--
+--     * If @'RestrictedVRFTiebreaker' maxDist@: Only do the VRF comparison (as
+--       in the previous step) if the slot numbers differ by at most @maxDist@.
+--
+-- === Non-transitivity of 'RestrictedVRFTiebreaker'
+--
+-- When using @cfg = 'RestrictedVRFTiebreaker' maxDist@, the chain order is not
+-- transitive. As an example, suppose @maxDist = 5@ and consider three
+-- 'PraosChainSelectView's with the same chain length and pairwise different
+-- issuers and, as well as
+--
+-- +------+---+---+---+
+-- |      | a | b | c |
+-- +======+===+===+===+
+-- | Slot | 0 | 3 | 6 |
+-- +------+---+---+---+
+-- | VRF  | 3 | 2 | 1 |
+-- +------+---+---+---+
+--
+-- Then we have @'preferCandidate' cfg a b@ and @'preferCandidate' b c@, but
+-- __not__ @'preferCandidate' a c@ (despite @a < c@).
+--
+-- === Rationale for the rules
+--
+-- 1. The abstract Consensus layer requires that we first compare based on chain
+--    length (see __Chain extension precedence__ in 'ChainOrder').
+--
+-- 2. Consider the scenario where the hot key of a block issuer was compromised,
+--    and the attacker is now minting blocks using that identity. The actual
+--    block issuer can use their cold key to issue a new hot key with a higher
+--    opcert issue number and set up a new pool. Due to this tiebreaker rule,
+--    the blocks minted by that pool will take precedence (allowing the actual
+--    block issuer to decide on the block contents), and they will end up on the
+--    honest chain quickly, which means that the adversary can't extend any
+--    chain containing such a block as it would violate the monotonicity
+--    requirement on opcert issue numbers.
+--
+--     See "3.7 Block Validity and Operational Key Certificates" in "Design
+--     Specification for Delegation and Incentives in Cardano" by Kant et al for
+--     more context.
+--
+-- 3. The main motivation to do VRF comparisons the usually final tiebreaker is
+--    to avoid the "Frankfurt problem":
+--
+--     With only the first two rules for the chain order, almost all blocks with
+--     equal block number are equally preferrable. Consider two block issuers
+--     minting blocks in very nearby slots. As we never change our selection
+--     from one chain to an equally preferrable one, the first block to arrive
+--     at another pool is the one to be adopted, and will be extended the next
+--     time the pool is elected if no blocks with a higher block number arrive
+--     in the meantime. This incentivizes block producers to concentrate
```

Comment:
> This incentivizes ...

Perhaps we should say "We observed that this effectively incentivizes ...", since it was never established whether a fully rational actor would actually be incentivized :)